openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java

```diff
@@ -436,20 +323,185 @@ private void getObjectives(Exercise duplicatedExercise, Exercise originalExercis
+  }
+
+  // -- SELECT --
+  private void select(
+      CriteriaBuilder cb,
+      CriteriaQuery<Tuple> cq,
+      Root<Exercise> exerciseRoot,
+      Map<String, Join<Base, Base>> joinMap) {
+    // Array aggregations
+    Join<Base, Base> exerciseTagsJoin = exerciseRoot.join("tags", JoinType.LEFT);
+    joinMap.put("tags", exerciseTagsJoin);
+    Expression<String[]> tagIdsExpression = arrayAggOnId(cb, exerciseTagsJoin);
+
+    Join<Base, Base> injectsJoin = exerciseRoot.join("injects", JoinType.LEFT);
+    joinMap.put("injects", injectsJoin);
+    Expression<String[]> injectIdsExpression = arrayAggOnId(cb, injectsJoin);
+    // SELECT
+    cq.multiselect(
+            exerciseRoot.get("id").alias("exercise_id"),
+            exerciseRoot.get("name").alias("exercise_name"),
+            exerciseRoot.get("status").alias("exercise_status"),
+            exerciseRoot.get("subtitle").alias("exercise_subtitle"),
+            exerciseRoot.get("category").alias("exercise_category"),
+            exerciseRoot.get("start").alias("exercise_start_date"),
+            exerciseRoot.get("updatedAt").alias("exercise_updated_at"),
+            tagIdsExpression.alias("exercise_tags"),
+            injectIdsExpression.alias("exercise_injects"))
+        .distinct(true);
+
+    // GROUP BY
+    cq.groupBy(Collections.singletonList(exerciseRoot.get("id")));
+  }
+
+  // -- EXECUTION --
+  private List<ExerciseSimple> execution(TypedQuery<Tuple> query) {
+    return query.getResultList().stream()
+        .map(
+            tuple -> {
+              ExerciseSimple exerciseSimple = new ExerciseSimple();
+              exerciseSimple.setId(tuple.get("exercise_id", String.class));
+              exerciseSimple.setName(tuple.get("exercise_name", String.class));
+              exerciseSimple.setStatus(tuple.get("exercise_status", ExerciseStatus.class));
+              exerciseSimple.setSubtitle(tuple.get("exercise_subtitle", String.class));
+              exerciseSimple.setCategory(tuple.get("exercise_category", String.class));
+              exerciseSimple.setStart(tuple.get("exercise_start_date", Instant.class));
+              exerciseSimple.setUpdatedAt(tuple.get("exercise_updated_at", Instant.class));
+              exerciseSimple.setTagIds(
+                  new HashSet<>(Arrays.asList(tuple.get("exercise_tags", String[].class))));
+              exerciseSimple.setInjectIds(tuple.get("exercise_injects", String[].class));
+              return exerciseSimple;
+            })
         .toList();
   }
 
-  public List<AtomicTestingMapper.ExpectationResultsByType> getGlobalResults(
-      @NotBlank String exerciseId) {
+  // -- COMPUTED ATTRIBUTES --
+  private void setComputedAttribute(List<ExerciseSimple> exercises) {
+    // -- MAP TO GENERATE TARGETSIMPLEs
+    Set<String> exerciseIds =
+        exercises.stream().map(ExerciseSimple::getId).collect(Collectors.toSet());
+
+    if (exerciseIds != null && !exerciseIds.isEmpty()) {
```

Comment:
```suggestion
    if (!exerciseIds.isEmpty()) {
```
exerciseIds cannot be null :)